src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import classNames from 'classnames';
+import { useState } from 'react';
+import { removeTodo, updateTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { NewTodoField } from './NewTodoField';
+
+type Props = {
+  todo: Todo,
+  loadTodos: () => void,
+  onSetDeleteTodoError: (isError: boolean) => void,
+  onSetUpdateTodoError: (isError: boolean) => void,
+  addTodoToLoadingList: (idToAdd: number) => void,
+  deleteTodoOfLoadingList: (idToAdd: number) => void,
+  loadingList: number[],
+};
+
+export const TodoInfo: React.FC<Props> = (
+  {
+    todo,
+    loadTodos,
+    onSetDeleteTodoError,
+    onSetUpdateTodoError,
+    addTodoToLoadingList,
+    deleteTodoOfLoadingList,
+    loadingList,
+  },
+) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const [newTitle, setNewTitle] = useState(title);
+
+  const [isEditing, setIsEditing] = useState(false);
+
+  const onDeleteTodo = async () => {
+    addTodoToLoadingList(id);
+
+    try {
+      await removeTodo(id);
+      await loadTodos();
+    } catch {
+      onSetDeleteTodoError(true);
+      setTimeout(() => {
+        onSetDeleteTodoError(false);
+      }, 3000);
+    }
+
+    deleteTodoOfLoadingList(id);
+  };
+
+  const onUpdateStatusTodo = async () => {
+    addTodoToLoadingList(id);
+
+    try {
+      await updateTodo(id, title, !completed);
+      await loadTodos();
+    } catch {
+      onSetUpdateTodoError(true);
+      setTimeout(() => {
+        onSetUpdateTodoError(false);
+      }, 3000);
+    }
+
+    deleteTodoOfLoadingList(id);
+  };
+
+  const handleDoubleClick = (event: React.MouseEvent<HTMLInputElement>) => {
+    if (event.detail === 2) {
+      setIsEditing(true);
+    }
+  };
+
+  const onUpdateTodoTitle = async () => {
+    if (title === newTitle) {
```

Comment:
also if newTitle is empty it must be deleted
![image](https://user-images.githubusercontent.com/113117393/208439792-32a2940f-c290-46f9-919d-74e332b0b563.png)